ocpp_codec/serializer.py

```diff
@@ -33,26 +34,30 @@ def _required_fields(dataclass_class) -> typing.List[dataclasses.Field]:
     return [field for field in fields(dataclass_class) if not _is_optional(field)]
 
 
-def _run_validator(field: dataclasses.Field, data: typing.Any, *, parsing: bool) -> typing.Any:
+def _clean_data(field: dataclasses.Field, data: typing.Any, *, parsing: bool) -> typing.Any:
     # Fetch the validators to run
     validator_list = field.metadata.get('validators', [validators.noop])
     # Defining a single validator is supported, turn it to a list for easier processing
     if not isinstance(validator_list, list):
         validator_list = [validator_list]
 
-    validated_data = data
     for validator in validator_list:
-        if isinstance(validator, validators.BaseEncoder):
-            # Pick the right method based on whether we're parsing a message, or serializing a field
-            validator = validator.from_json if parsing else validator.to_json
-
         try:
-            validated_data = validator(data)
+            validator(data)
         except errors.BaseOCPPError as exc:
             # Inject field data in exception, as the validator doesn't know the field's name
             exc.details.setdefault('field', field.name)
             raise exc
-    return validated_data
+
+    # Apply the defined encoder if it exists
+    encoder = field.metadata.get('encoder', encoders.NoopEncoder())
```

Comment:
Why calling the `NoopEncoder` instead of returning early if there isn't one ?

Reply:
the idea was to keep a single point of exit in the function, but I think this might be so simple that I actually over complicate it with this class... I'll drop this, since I'm also dropping `valodators.noop` now that I just return an empty list of validators if none were defined.